NumberBaseball/NumberBaseball/main.swift

```diff
@@ -1,10 +1,77 @@
 //
 //  NumberBaseball - main.swift
-//  Created by yagom. 
+//  Created by yagom.
 //  Copyright © yagom academy. All rights reserved.
-// 
+//
 
 import Foundation
 
-print("Hello, World!")
+var repeatCheck = true
+
+func generateNumbers() -> Set<Int> {
+    var randomNumber = Set<Int>()
+    
+    while randomNumber.count < 3 {
+        randomNumber.insert(Int.random(in: 1...9))
+    }
+    return randomNumber
+}
+
+func checkStrikeOrBall(answerNumber: Array<Int>, tryCount: Int) -> [String : Int] {
+    let suggestNumber = Array<Int>(generateNumbers())
+    var strikeCount = 0
+    var ballCount = 0
+    
+    if answerNumber[0] == suggestNumber[0] {
+        strikeCount += 1
+    } else if suggestNumber.contains(answerNumber[0]) {
+        ballCount += 1
+    }
+    if answerNumber[1] == suggestNumber[1] {
+        strikeCount += 1
+    } else if suggestNumber.contains(answerNumber[1]) {
+        ballCount += 1
+    }
+    if answerNumber[2] == suggestNumber[2] {
+        strikeCount += 1
+    } else if suggestNumber.contains(answerNumber[2]) {
+        ballCount += 1
+    }
+    
+    print("임의의 수:\(suggestNumber)")
+    print("\(strikeCount) 스트라이크, \(ballCount) 볼")
+    print("남은기회 : \(tryCount)")
+    if strikeCount == 3 {
+        print("인간승리...!")
+    }else if tryCount == 0 {
+        print("컴퓨터 승리...!")
+    }
+    
+    var strikeBallCount: [String : Int] = [:]
+    strikeBallCount["strike"] = strikeCount
+    strikeBallCount["trycount"] = tryCount
+    
+    return strikeBallCount
+}
+
+func gameStart() {
+    var answerNumber = Set<Int>()
+    var tryCount = 9
+    
+    while answerNumber.count < 3 {
+        answerNumber.insert(Int.random(in: 1...9))
+    }
+    while repeatCheck {
+        tryCount -= 1
+        let strikeBallCount = checkStrikeOrBall(answerNumber: Array<Int>(answerNumber), tryCount: tryCount)
```

Comment:
```suggestion
        let strikeBallCount = checkStrikeOrBall(answerNumber: answerNumber, tryCount: tryCount)
```

매개변수로 전달하면서 형변환 하는건 조금 이상한거 같아요
형변환을 미리 한 후 전달해주는게 좋지 않을까요??

추가로 answerNumber는 Set or Int 배열인데, 복수로 써야할것 같습니다
네이밍도 현재는 정답숫자로 읽히는데, 정답..? 보다는 컴퓨터가 생성한 숫자라는 뜻이 더 잘 어울리지 않을까요?

Reply:
answerNumber 변수명을 randomNumberGeneratedByComputer 으로 변경하였습니다